src/pages/notifications/index.vue

```diff
@@ -306,18 +306,23 @@ export default {
       }
     }
     this.$api.$post(updateEventLastSeen());
-    this.lastEntryTs = Date.now();
+    this.lastUpdatedTime = Date.now();
   },
   // For SPA navigation
   beforeRouteLeave(to, from, next) {
-    this.updateEventLastSeenTs(this.lastEntryTs || Date.now());
+    if (this.lastUpdatedTime) {
+      this.updateEventLastSeenTs(this.lastUpdatedTime);
+    }
     next();
   },
 
   methods: {
     ...mapActions(['fetchWalletEvents', 'updateEventLastSeenTs']),
     async handleRefresh() {
+      this.updateEventLastSeenTs(this.lastUpdatedTime);
       await this.fetchWalletEvents();
+      this.$api.$post(updateEventLastSeen());
```

Comment:
can we check if there's actually new wallet event? if not we don't need to call update